File: cpp/1775. 通过最少操作次数使数组的和相等.cpp
class Solution {
public:
    int minOperations(vector<int>& nums1, vector<int>& nums2) {
       int diff=0;
       int m=nums1.size(),n=nums2.size();
       if(6*m<n||6*n<m) return -1;
       diff=accumulate(nums1.begin(),nums1.end(),0)-accumulate(nums2.begin(),nums2.end(),0);
       if(diff<0){
           diff=-diff;
           swap(nums1,nums2);
       }
       vector<int> cnt(6,0);
       for(auto x:nums1) cnt[x-1]++;
       for(auto x:nums2) cnt[6-x]++;
       int ans=0;
       for(int i=5;i>0;i--){
           if(i*cnt[i]>=diff){
               return ans+(diff+i-1)/i;
           }
           diff-=i*cnt[i];
           ans+=cnt[i];
       }
       return ans;
    }
};

File: 刷题/golang/1775. 通过最少操作次数使数组的和相等.go
func minOperations(nums1 []int, nums2 []int) int {
    if 6*len(nums1) < len(nums2) ||6*len(nums2)<len(nums1){
        return -1
    }
    d:=0
    for _,x:=range nums2 {
        d+=x
    }
    for _,x:=range nums1 {
        d-=x
    }
    if d<0{
        d=-d
        nums1,nums2=nums2,nums1
    }
    cnt:=[6]int{}
    for _,x:=range nums1{
        cnt[6-x]++
    }
    for _,x:=range nums2{
        cnt[x-1]++
    }
    ans:=0
    for i:=5;i>0;i--{
        if i*cnt[i]>=d{
            return ans+(d+i-1)/i
        }
        ans+=cnt[i]
        d-= i*cnt[i]
    }
    return ans
}